### PR TITLE
feat: [PIE-4572]: Added Save Pipelines with errors for update

### PR DESCRIPTION
**Pull request by smjt-h**
## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [x] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/35670)

### CODE REVIEW

File: 800-pipeline-service/src/main/java/io/harness/pms/pipeline/service/PMSPipelineServiceImpl.java [hunk 1]
```diff
@@ -253,19 +253,31 @@ public Optional<PipelineEntity> getWithoutPerformingValidations(
...
         pipelineEntity.getOrgIdentifier(), pipelineEntity.getProjectIdentifier(), pipelineEntity.getIdentifier());
     GovernanceMetadata governanceMetadata = pmsPipelineServiceHelper.validatePipelineYaml(pipelineEntity);
     if (governanceMetadata.getDeny()) {
       return PipelineCRUDResult.builder().governanceMetadata(governanceMetadata).build();
     }
+    PipelineEntity updatedEntity = updatePipelineWithoutValidation(pipelineEntity, changeType);
+    return PipelineCRUDResult.builder().governanceMetadata(governanceMetadata).pipelineEntity(updatedEntity).build();
+  }
+
+  public PipelineEntity updatePipelineWithoutValidation(PipelineEntity pipelineEntity, ChangeType changeType) {
```
Comment (namanvermah) [inline 800-pipeline-service/src/main/java/io/harness/pms/pipeline/service/PMSPipelineServiceImpl.java:272]: Why is this method public?

File: 800-pipeline-service/src/main/java/io/harness/pms/pipeline/service/PMSPipelineServiceImpl.java [hunk 2]
```diff
@@ -253,19 +253,31 @@ public Optional<PipelineEntity> getWithoutPerformingValidations(
 
   @Override
   public PipelineCRUDResult updatePipelineYaml(PipelineEntity pipelineEntity, ChangeType changeType) {
+    if (pipelineEntity.getIsDraft() != null && pipelineEntity.getIsDraft()) {
```
Comment (namanvermah) [inline 800-pipeline-service/src/main/java/io/harness/pms/pipeline/service/PMSPipelineServiceImpl.java:256]: Please write unit test for this if block

**Review by namanvermah — CHANGES_REQUESTED** [review]

**Review by archit-harness — APPROVED** [review]
> Reviewed 1 of 1 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35670)* status: 1 file(s) to review. 1 change requests. 1 of 2 approvals obtained.

File: 800-pipeline-service/src/main/java/io/harness/pms/pipeline/service/PMSPipelineServiceImpl.java (hunk 1 above)
Comment (smjt-h) [reply 800-pipeline-service/src/main/java/io/harness/pms/pipeline/service/PMSPipelineServiceImpl.java:272]: made private

File: 800-pipeline-service/src/main/java/io/harness/pms/pipeline/service/PMSPipelineServiceImpl.java (hunk 2 above)
Comment (smjt-h) [reply 800-pipeline-service/src/main/java/io/harness/pms/pipeline/service/PMSPipelineServiceImpl.java:256]: done

**Review by archit-harness — DISMISSED** [review]
> Reviewed 2 of 2 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35670)* status: 2 file(s) to review. 1 change requests. 2. unresolved discussions. 1 of 2 approvals obtained.

**Review by archit-harness — APPROVED** [review]
> Reviewed 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35670)* status: 2 file(s) to review. 1 change requests. 2. unresolved discussions. 1 of 2 approvals obtained.

**Review by namanvermah — APPROVED** [review]
> Reviewed 1 of 2 files at r2, 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35670)* status: :shipit: complete! ✓

**Pull request merged**